packages/@aws-cdk/aws-ec2-alpha/lib/route.ts

```diff
@@ -331,9 +369,15 @@ export class NatGateway extends Resource implements IRouteTarget {
     this.connectivityType = props.connectivityType || NatConnectivityType.PUBLIC;
     this.maxDrainDuration = props.maxDrainDuration || Duration.seconds(350);
 
+    if (this.connectivityType === NatConnectivityType.PUBLIC) {
+      if (!props.vpc && !props.allocationId) {
+        throw new Error('Either provide vpc or allocationId');
+      }
+    }
+
     // If user does not provide EIP, generate one for them
     var aId: string | undefined;
-    if (this.connectivityType == 'public') {
+    if (this.connectivityType == NatConnectivityType.PUBLIC) {
```

Comment:
`===`

Reply:
i see it updated in the last revision, not sure if you;re still seeing the same https://github.com/aws/aws-cdk/pull/31224/files/5c7548c4954c8b3433cef4b45e7f0955825ddf1b#diff-65117dd44ab7b3f0b0f8d0421973a71b104cd76b4dc5147237da60616c718b2fR372